hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/lifecycle/boot/ConnectivityChecker.java

```diff
@@ -0,0 +1,70 @@
+package com.blackducksoftware.integration.hub.detect.lifecycle.boot;
+
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.blackducksoftware.integration.hub.detect.DetectInfo;
+import com.blackducksoftware.integration.hub.detect.configuration.DetectConfiguration;
+import com.blackducksoftware.integration.hub.detect.exception.DetectUserFriendlyException;
+import com.blackducksoftware.integration.hub.detect.exitcode.ExitCodeType;
+import com.blackducksoftware.integration.hub.detect.help.DetectOptionManager;
+import com.blackducksoftware.integration.hub.detect.workflow.event.EventSystem;
+import com.blackducksoftware.integration.hub.detect.workflow.phonehome.OnlinePhoneHomeManager;
+import com.blackducksoftware.integration.hub.detect.workflow.phonehome.PhoneHomeManager;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.synopsys.integration.blackduck.api.generated.discovery.ApiDiscovery;
+import com.synopsys.integration.blackduck.api.generated.response.CurrentVersionView;
+import com.synopsys.integration.blackduck.configuration.BlackDuckServerConfig;
+import com.synopsys.integration.blackduck.phonehome.BlackDuckPhoneHomeHelper;
+import com.synopsys.integration.blackduck.service.BlackDuckService;
+import com.synopsys.integration.blackduck.service.BlackDuckServicesFactory;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+public class ConnectivityChecker {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public ConnectivityResult determineConnectivity(DetectConfiguration detectConfiguration, DetectOptionManager detectOptionManager, DetectInfo detectInfo, Gson gson, ObjectMapper objectMapper, EventSystem eventSystem)
+        throws DetectUserFriendlyException {
+
+        logger.info("Detect will check if it can communicate with the Black Duck Server.");
+        Slf4jIntLogger blackduckLogger = new Slf4jIntLogger(logger);
+        BlackDuckServerConfig blackDuckServerConfig = detectOptionManager.createBlackduckServerConfig();
+
+        boolean connected;
+        logger.info("Attempting connection to the Black Duck server");
+        if (blackDuckServerConfig.canConnect(blackduckLogger)) {

```

Comment:
I think you want a brief
`if (!blackDuckServerConfig.canConnect(blackduckLogger)) {
  logger.error("Failed to connect to the Black Duck server")
  return ConnectivityResult.failure("Could not reach the Black Duck server or the credentials were invalid.");
}
`

Then you don't need the connected variable, you lose the else branch, and the long chunk of code moves to the left, AND it becomes clearer what you do when you don't connect.

The principle is to fail fast at the beginning and get out of your code if you can - you don't want to carry around the weight of decisions that have already been made for any longer than you have to.